### PR TITLE
Change Local Storage `set-default` to `default`

### PR DESCRIPTION
**Pull request by mateoflorido**
## Overview
Replace the Local Storage `set-default` config occurrences to `default`.

### CODE REVIEW

**Comment by ktsakalozos** [comment]
I think there should be a docs update too https://github.com/canonical/k8s-snap/blob/main/docs/src/howto/storage.md

**Review by louiseschmidtgen — COMMENTED** [review]
Thanks for updating the naming, this is cleaning things up nicely! Glad we won't do `GetSetDefault`anymore :)

File: src/k8s/api/v1/cluster_config.go [hunk 1]
```diff
@@ -77,13 +77,13 @@ type LocalStorageConfig struct {
 	Enabled       *bool   `json:"enabled,omitempty" yaml:"enabled,omitempty"`
 	LocalPath     *string `json:"local-path,omitempty" yaml:"local-path,omitempty"`
 	ReclaimPolicy *string `json:"reclaim-policy,omitempty" yaml:"reclaim-policy,omitempty"`
-	SetDefault    *bool   `json:"set-default,omitempty" yaml:"set-default,omitempty"`
+	Default       *bool   `json:"default,omitempty" yaml:"default,omitempty"`
 }
```
Comment (neoaggelos) [inline src/k8s/api/v1/cluster_config.go:81]: Just to note: this is a breaking change in the API. We do not have stable releases yet, so we can do it.

File: src/k8s/pkg/k8sd/types/cluster_config_features.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ type LocalStorage struct {
 	Enabled       *bool   `json:"enabled,omitempty"`
 	LocalPath     *string `json:"local-path,omitempty"`
 	ReclaimPolicy *string `json:"reclaim-policy,omitempty"`
-	SetDefault    *bool   `json:"set-default,omitempty"`
+	Default       *bool   `json:"default,omitempty"`
 }
```
Comment (neoaggelos) [inline src/k8s/pkg/k8sd/types/cluster_config_features.go:39]: Of note, this `json:"set-default"` -> `json:"default"` is also a breaking change, which would make the value `nil` in existing clusters.

In the future, we could keep the `json:"set-default"` around, and just rename the struct field.

But again, seeing as there are no stable releases yet, we're good to go with the change.

**Review by neoaggelos — APPROVED** [review]
LGTM, nice work @mateoflorido. Adding a couple of notes regarding the breaking changes contained in the PR, this is something we will not be able to do once we get a stable release out the door

**Review by bschimke95 — APPROVED** [review]
LGTM, great work

**Pull request merged**